homeassistant/components/media_player/denonavr.py

```diff
@@ -27,8 +30,19 @@
 DEFAULT_NAME = None
 DEFAULT_SHOW_SOURCES = False
 DEFAULT_TIMEOUT = 2
+DEFAULT_SOUND_MODE = True
+DEFAULT_SOUND_MODE_DICT = OrderedDict([('MUSIC', 'PLII MUSIC'),
+                                       ('MOVIE', 'PLII MOVIE'),
+                                       ('GAME', 'PLII GAME'),
+                                       ('PURE DIRECT', 'DIRECT'),
+                                       ('AUTO', 'None'),
+                                       ('DOLBY DIGITAL', 'DOLBY DIGITAL'),
+                                       ('MCH STEREO', 'MULTI CH STEREO'),
+                                       ('STEREO', 'STEREO')])
```

Comment:
Well, for now it needs to be statically defined because I could not figure out how to get that data from the receiver. At least it is not in the XML file that you can request from the receiver.
However in the official phone aplication of Marantz you can also change the sound mode and their is a list of sound modes their and it nows wich sound mode is currently set. Since this phone application is generic for all Marantz Receivers (that support ethernet), their must be a way to get this information. I am pretty sure the phone application uses normal http requests just like the build in web server does.

Theirfor it is probably possible, I just do not know how.
And at the moment I do not have enough time to figure this out.

It was already fairly difficult to get the whole dictionary working as a configurable option, because the voluptuous checker was not to happy with the syntax.
So for now I think we have to settle for this, but in the future it would definetly be a greath improvement to get this working using some sort of request to the receiver.

Reply:
Ok, I think it is fine (for now) ho hardcode that, although I think that the proper place for this is also inside denonavr library rather than here.

I understand that it can be quite hard to decipher what is going on, but assuming you want to look into it you may want to run tcpdump on your router to capture the traffic for later analysis with wireshark.